integrationTests/state/stateTrie/stateTrie_test.go

```diff
@@ -1838,3 +1839,140 @@ func checkCodeConsistency(
+	integrationTests.CreateMintingForSenders(nodes, senderShard, sendersPrivateKeys, valMinting)
+
+	round = integrationTests.IncrementAndPrintRound(round)
+	nonce++
+	round, nonce = integrationTests.ProposeAndSyncOneBlock(t, nodes, idxProposers, round, nonce)
+
+	shardNode := nodes[0]
+
+	dataTriesRootHashes, codeMap := generateAccounts(shardNode, accounts)
+
+	_, _ = shardNode.AccntState.Commit()
+	round, nonce = integrationTests.ProposeAndSyncOneBlock(t, nodes, idxProposers, round, nonce)
+
+	numAccountsToRemove := 2
+	roundsToWait := 50
+
+	removedAccounts := make(map[int]struct{})
+	for i := 0; i < roundsToWait; i++ {
+		for j := 0; j < numAccountsToRemove; j++ {
+			accountIndex := rand.Intn(nrAccounts)
+			removedAccounts[accountIndex] = struct{}{}
+			account, err := shardNode.AccntState.GetExistingAccount(accounts[accountIndex])
+			if err != nil {
+				continue
+			}
+			code := account.(state.UserAccountHandler).GetCode()
+
+			_ = shardNode.AccntState.RemoveAccount(account.AddressBytes())
+
+			codeMap[string(code)]--
+		}
+
+		_, _ = shardNode.AccntState.Commit()
+		round, nonce = integrationTests.ProposeAndSyncOneBlock(t, nodes, idxProposers, round, nonce)
+		checkCodeConsistency(t, shardNode, codeMap)
+	}
+
+	delayRounds := 5
+	for i := 0; i < delayRounds; i++ {
+		round, nonce = integrationTests.ProposeAndSyncOneBlock(t, nodes, idxProposers, round, nonce)
+	}
+
+	checkDataTrieConsistency(t, shardNode.AccntState, removedAccounts, dataTriesRootHashes)
+}
+
+func generateAccounts(
+	shardNode *integrationTests.TestProcessorNode,
+	accounts [][]byte,
+) ([][]byte, map[string]int) {
+	numCodes := 100
+	codeMap := getCodeMap(numCodes)
+	codeArray := make([][]byte, 0)
+	for code := range codeMap {
+		codeArray = append(codeArray, []byte(code))
+	}
+
+	dataTriesRootHashes := make([][]byte, 0)
+	dataTrieSize := 5
+	for i := 0; i < len(accounts); i++ {
+		account, _ := shardNode.AccntState.LoadAccount(accounts[i])
+
+		code := codeArray[rand.Intn(numCodes)]
+		account.(state.UserAccountHandler).SetCode(code)
+		codeMap[string(code)]++
+
+		for j := 0; j < dataTrieSize; j++ {
+			account.(state.UserAccountHandler).DataTrieTracker().SaveKeyValue(getDataTrieEntry())
+		}
+
+		_ = shardNode.AccntState.SaveAccount(account)
+
+		rootHash := account.(state.UserAccountHandler).GetRootHash()
+		dataTriesRootHashes = append(dataTriesRootHashes, rootHash)
+	}
+
+	return dataTriesRootHashes, codeMap
+}
+
+func getDataTrieEntry() ([]byte, []byte) {
+	index := strconv.Itoa(rand.Intn(math.MaxInt32))
+	key := []byte("key" + index)
+	value := []byte("value" + index)
+
+	return key, value
+}
+
+func checkDataTrieConsistency(
+	t *testing.T,
+	adb state.AccountsAdapter,
+	removedAccounts map[int]struct{},
+	dataTriesRootHashes [][]byte,
+) {
+	for i, rootHash := range dataTriesRootHashes {
+		_, ok := removedAccounts[i]
+		if ok {
+			err := adb.RecreateTrie(rootHash)
+			assert.NotNil(t, err)
+		} else {
+			err := adb.RecreateTrie(rootHash)
```

Comment:
does this `RecreateTrie` affect the state of the object in the future uses? If yes, make a copy of it before checking that recreate trie does not error

Reply:
There is no need for a copy here.